portality/regex.py

```diff
@@ -1,16 +1,39 @@
 import re
 
-DOI = r"^((https?://)?((dx\.)?doi\.org/|hdl\.handle\.net/)|doi:|info:doi/|info:hdl/)?(?P<id>10\.\S+/\S+)$"
-DOI_COMPILED = re.compile(DOI, re.IGNORECASE)
-ORCID = r"^https://orcid\.org/[0-9]{4}-[0-9]{4}-[0-9]{4}-\d{3}[\dX]$"
+from lxml import etree
+
+from portality.core import app
+
+DOI_XPATH = '/xs:schema[@version="1.2"]/xs:complexType[@name="recordType"]/xs:sequence/xs:element[@name="doi"]/xs:simpleType/xs:restriction/xs:pattern/@value'
```

Comment:
I don't think specifying all the way from the root is the best way to do this - not least because we'll be changing XSD version in future and we don't want to change our code with that. I recommend using a `find` which lets you use a more minimal XPATH.